### PR TITLE
adding more themes for user. 

### PR DESCRIPTION
**Pull request by Shivank98**
I thought it's good to have some more themes for the user. currently, I just added a  theme.html which manually can change the theme. 
I also created a field `user_theme` though I am not using that right now. 
Should we keep it this way only( having user_theme) or there is some better way available?

### CODE REVIEW

**Comment by theSage21** [comment]
I'd rather not touch the database for this info since it's not sensitive. A couple of ideas:

- Store user choice in cookies. Switch css based on cookie values on browser
- Ask them to use greasemonkey to switch themes. I'd prefer this since it's a zero effort thing from our side.

**Comment by Shivank98** [comment]
True, touching DB is not a nice idea. will do it with cookies.

**Comment by theSage21** [comment]
That sounds pretty good. Instead of cookies you might want to use [browser local storage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage). That way you could simply store the entire stylesheet in the storage and just switch based on user options.

**Comment by Shivank98** [comment]
no idea on it [browser local storage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage). can you point me to some specific so I can look into that?

**Comment by theSage21** [comment]
Sure sure. I just added you to a private project. Once you accept the invitation you can see how to [get from local storage](https://github.com/theSage21/orfik/blob/master/templates/base.html#L19-L20) and [how to set local storage](https://github.com/theSage21/orfik/blob/master/templates/index.html#L10-L11).

This is equivalent to what you might have done using something like [js-cookie](https://github.com/js-cookie/js-cookie).

**Comment by Shivank98** [comment]
I tried to do it using some js function but that seems to work on the current page only.not on all website.

File: views/base.html [hunk 1]
```diff
@@ -3,7 +3,12 @@
    <head>
        {% include 'meta.html' %}
       <title>SoA</title>
-      <link rel="stylesheet" href="https://unpkg.com/sakura.css/css/sakura.css" type="text/css">
+      <script type="text/javascript">
+        function swapStyleSheet(sheet){
+             document.getElementById('pagestyle').setAttribute('href',sheet);
```
Comment (theSage21) [inline views/base.html:8]: Instead of just changing the href shouldn't we be creating a full new element and replacing the existing one?

Comment (Shivank98) [reply views/base.html:8]: Yeah, I looked there but couldn't get a lot out of it. 
Ohk, let me see how we can create new element instead of replacing.

Comment (Shivank98) [reply views/base.html:8]: even sakura demo isn't managing the user theme. it comes to normal if you move to another page or reload.

**Comment by theSage21** [comment]
Oh accha. In that case the simplest thing would be to use a cookie variable
yo set theme option and load different CSS base on that in jinja.

**Comment by Shivank98** [comment]
Now it works.

File: views/base.html [hunk 2]
```diff
@@ -1,29 +1,73 @@
...
+        <title>SoA</title>
+        <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>
+        <link id="pagestyle" rel="stylesheet" href="https://unpkg.com/sakura.css/css/sakura.css" type="text/css" media="screen">
+        <!--
+            <script type="text/javascript">
+              function swapStyleSheet(sheet){
+                   document.getElementById('pagestyle').setAttribute('href',sheet);
+            
+              -->
+        <p>
```
Comment (theSage21) [inline views/base.html:14]: let's not include p tags inside head. If possible I'd like to move these options html to the accounts page. Perhaps a subheading of "This device"?

File: views/base.html [hunk 3]
```diff
@@ -1,29 +1,73 @@
...
+            Choose theme: 
+            <select>
+                <option selected="selected" value="sakura">Default</option>
+                <option value="sakura-dark">Night</option>
+                <option value="sakura-earthly">Earth</option>
+                <option value="sakura-vader">Vampire</option>
+            </select>
+        </p>
+        <script>
```
Comment (theSage21) [inline views/base.html:23]: Can we move this code to `themes.html` and use `{% include 'themes.html %}` in base? If possible I'd like to keep base as clean as possible.

Comment (Shivank98) [reply views/base.html:23]: In account's Html file or theme.html?  both suggestions are different. or you want to just show it in account's HTML but using `{ % include 'themes.html' %}`

**Comment by theSage21** [comment]
I was thinking ki the choice of switching themes can be done in the
accounts page.

Since the theme applying logic will be needed on all pages so we can
include theme.html in base.html

**Comment by Shivank98** [comment]
Is this what do you want to say?

**Comment by theSage21** [comment]
Yeah this makes a little more sense. If you clean up the base html so that the diff only shows the include statement this should be ready to merge.

**Comment by Shivank98** [comment]
Do you mean removing commented part?

**Comment by theSage21** [comment]
Yeah. That and ask the other things which are shown in the diff.

**Comment by Shivank98** [comment]
that's what I could figure out from git diff

**Comment by theSage21** [comment]
I've made some changes in https://github.com/Shivank98/Summer-of-Algorithm/pull/2

**Pull request merged**